packages/database/src/migrations/20211019223410-ConvertFanfanaOlaHpuOverlaysToReportServer-modifies-data.js

```diff
@@ -0,0 +1,264 @@
+        color: 'orange',
+        value: 'School',
+      },
+      {
+        icon: 'circle',
+        name: 'Workplace',
+        color: 'pink',
+        value: 'Workplace',
+      },
+      {
+        icon: 'circle',
+        name: 'Community',
+        color: 'lightblue',
+        value: 'Community',
+      },
+      {
+        icon: 'circle',
+        name: 'Multiple Settings',
+        color: 'yellow',
+        value: 'NO_UNIQUE_VALUE',
+      },
+      {
+        icon: 'circle',
+        name: 'No data',
+        color: 'Grey',
+        value: 'null',
+      },
+    ],
+    measureConfig: {
+      Church: {
+        name: 'Church Screenings',
+        sortOrder: 1,
+      },
+      Community: {
+        name: 'Community Screenings',
+        sortOrder: 2,
+      },
+      School: {
+        name: 'School Screenings',
+        sortOrder: 3,
+      },
+      Workplace: {
+        name: 'Workplace Screenings',
+        sortOrder: 4,
+      },
+      Total: {
+        name: 'Total Screenings',
+        sortOrder: 0,
+      },
+      $all: {
+        type: 'popup-only',
+        values: [
+          {
+            value: null,
+            hideFromPopup: true,
+          },
+        ],
+      },
+    },
+    displayType: 'icon',
+    hideFromPopup: true,
+  },
+  country_codes: '{"TO"}',
+  project_codes: '{"fanafana"}',
+});
+
+const addMapOverlay = async (db, permissionGroupId, mapOverlayGroupId, config) => {
+  const { reportCode, dataElements, code, dataGroups, name, sortOrder } = config;
+  const report = getReportObject(reportCode, dataElements, dataGroups);
+  const mapOverlayId = generateId();
+  const mapOverlay = getOverlayObject(reportCode, code, mapOverlayId, name);
+  await insertObject(db, 'report', {
+    id: generateId(),
+    permission_group_id: permissionGroupId,
+    ...report,
+  });
+  await insertObject(db, 'map_overlay', mapOverlay);
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: mapOverlayGroupId,
+    child_id: mapOverlayId,
+    child_type: 'mapOverlay',
+    sort_order: sortOrder,
+  });
+};
+
+const removeMapOverlay = (db, reportCode, overlayId) => {
+  return db.runSql(`
+    DELETE FROM "report" WHERE "code" = '${reportCode}';
+    DELETE FROM "map_overlay" WHERE "id" = '${overlayId}';
+    DELETE FROM "map_overlay_group_relation" WHERE "child_id" = '${overlayId}';
+  `);
+};
+
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  // get Permission Group Id and overlay group id
+  const permissionGroupId = await nameToId(db, 'permission_group', PERMISSION_GROUP);
+  const mapOverlayGroupId = await codeToId(db, 'map_overlay_group', HPU_GROUP_CODE);
+  // Remove legacy overlay relations
+  for (const code of LEGACY_OVERLAY_CODES) {
+    await db.runSql(`
+    DELETE FROM "map_overlay_group_relation" WHERE "child_id" = '${code}';
```

Comment:
This isn't gonna quite work because the `child_id` field now correctly refers to the `id` field of the `map_overlay` so you'll need to add a `codeToId` somewhere in this process